node/src/daemon/setup_reporter.rs

```diff
@@ -2413,9 +2413,18 @@ mod tests {
             "config_file_not_specified_but_exists",
         );
         {
-            let config_file_path = data_directory.join("config.toml");
-            let mut config_file = File::create(config_file_path).unwrap();
-            config_file.write_all(b"gas-price = \"10\"\n").unwrap();
+            let config_file_path = data_directory
```

Comment:
You could create a variable `data_dir_path` and then you could use it twice, in defining the config-file path and also for supply to the `create_dir_all()` fn.   

`config_file_path` would look like `config_file_path = data_dir_path.join("config.toml");